sandbox/src/main/java/org/robolectric/internal/bytecode/ClassInstrumentor.java

```diff
@@ -229,15 +229,19 @@ private boolean isJacocoInstrumented(MethodNode ctor) {
       if ((node instanceof LdcInsnNode && ((LdcInsnNode) node).cst instanceof ConstantDynamic)) {
         ConstantDynamic cst = (ConstantDynamic) ((LdcInsnNode) node).cst;
         return cst.getName().equals("$jacocoData");
-      } else if (insns.length > 1
-          && insns[0] instanceof LabelNode
-          && insns[1] instanceof MethodInsnNode) {
-        return "$jacocoInit".equals(((MethodInsnNode) insns[1]).name);
       }
     }
     return false;
   }
 
+  private boolean isJacocoInstrumentedInConstructorCallSuper(InsnList ctor) {
```

Comment:
Why is this a separate method? Why not just use `isJacocoInstrumented`? FYI `isJacocoInstrumented` is only designed for constructors.

Reply:
Yeap, that's no need. Only one `isJacocoInstrumented` detection is  need to check if a method (instead of `InsnList`) is instrumented.